vignettes/ipums-api.Rmd

```diff
@@ -0,0 +1,325 @@
+---
+title: "Interacting with the IPUMS microdata extract API"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{ipums-api}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+The IPUMS microdata extract API allows registered IPUMS users to define 
+extracts, submit extract requests, and download extract files without visiting 
+the IPUMS website. ipumsr includes functions that make it easy for R users to 
+interact with the extract API from their R session.
+
+```{r setup}
+library(ipumsr)
+library(dplyr) # not necessary to use API functions, but used in some examples
+library(purrr) # not necessary to use API functions, but used in some examples
+```
+
+# Setting up your API key
+
+**Add notes about how to create an API key here.**
+
+Once you've created an API key, you can choose to supply it as a function 
+argument whenever interacting with the API, or you can set the value of the 
+`IPUMS_API_KEY` environment variable to your key. One convenient way to set this 
+environment variable is to add code like the following to a file named 
+".Renviron" in your project directory or user home directory:
+
+```
+IPUMS_API_KEY = "paste-your-key-here"
+```
+
+If you use this approach with a project that uses Git for version control, be 
+sure to add the ".Renviron" file to ".gitignore" so that your API key is not 
+shared when someone clones your repository.
+
+
+# Defining your extract
+
+The `define_extract()` function returns an object of class "ipums_extract" which
+can then be submitted using the `submit_extract()` function.
+
+```{r eval=TRUE}
+extract_definition <- define_extract(
+  collection = "usa",
+  description = "Extract for API vignette",
+  samples = c("us2018a","us2019a"),
+  variables = c("AGE","SEX","RACE","STATEFIP"),
+  data_format = "fixed_width",
+  data_structure = "rectangular",
+  rectangular_on = "P"
+)
+```
+
+The `data_format`, `data_structure`, and `rectangular_on` arguments have default 
+values which match the values specified in the example above, and you can omit 
+the argument names as long as you maintain the proper order, so you could define 
+the same extract with:
+
+```{r}
+extract_definition <- define_extract(
+  "usa",
+  "Extract for API vignette",
+  c("us2018a","us2019a"),
+  c("AGE","SEX","RACE","STATEFIP")
+)
+```
+
+
+# Submitting your extract
+
+Submitting your extract is as easy as:
```

Comment:
I also slip into this, but I think in general it's better to avoid using this language of "easy" in technical documentation (you're not really convincing anyone that it's easy, and if someone has spent all day fighting with R, it can be infuriating). 

Trying to stay neutral:
> You can submit the extract by:

More explanation here: https://justsimply.dev

Reply:
Fixed, thanks!